prebuilt-ui/basic-embed/components/PrebuiltCall.js

```diff
@@ -0,0 +1,254 @@
+
+export default function PrebuiltCall(props) {
+  const [demoState, setDemoState] = useState('home');
+  const [isError, setIsError] = useState(false);
+  const [roomURL, setRoomURL] = useState('');
+  const [exp, setExp] = useState();
+  const [secs, setSecs] = useState();
+  const [roomValidity, setRoomValidity] = useState(false);
+  const roomURLRef = useRef(null);
+  const iframeRef = useRef(null);
+  const callFrame = useRef(null);
+  const [linkCopied, setLinkCopied] = useState(false);
+
+  // Updates the time left that displays in the UI
+  useEffect(() => {
+    if (!exp) {
+      return false;
+    }
+    const i = setInterval(() => {
+      const timeLeft = Math.floor((new Date(exp * 1000) - Date.now()) / 1000);
+      setSecs(`${Math.floor(timeLeft / 60)}:${`0${timeLeft % 60}`.slice(-2)}`);
+    }, 1000);
+    return () => clearInterval(i);
+  }, [exp]);
+
+  // Listens for a "call" demo state, and creates then joins a callFrame as soon as that happens
+  useEffect(() => {
+    if (!iframeRef?.current || demoState !== 'call') {
+      return;
+    }
+    try {
+      callFrame.current = DailyIframe.createFrame(iframeRef.current, {
+        showLeaveButton: true,
+        iframeStyle: {
+          height: '100%',
+          width: '100%',
+          aspectRatio: 16 / 9,
+          minwidth: '400px',
+          maxWidth: '920px',
+          border: '0',
+          borderRadius: '12px',
+        },
+      });
+      callFrame.current.join({
+        url: roomURL,
+      });
+    } catch (e) {
+      setDemoState('home');
+      setIsError(true);
+      return;
+    }
+
+    const handleLeftMeeting = () => {
+      setDemoState('home');
+      setRoomValidity(false);
+      callFrame.current.destroy();
+    };
+
+    callFrame.current.on('left-meeting', handleLeftMeeting);
+  }, [demoState, iframeRef, roomURL]);
+
+  const createRoom = useCallback(
+    async (e) => {
+      if (!roomURLRef?.current.value) {
+        const roomExp = Math.round(Date.now() / 1000) + 60 * 5;
+        try {
+          const res = await fetch('/api/room', {
+            method: 'POST',
+            headers: {
+              Accept: 'application/json',
+              'Content-Type': 'application/json',
+            },
+            body: JSON.stringify({
+              properties: {
+                roomExp,
+              },
+            }),
+          });
+          const roomJson = await res.json();
+          const { url } = roomJson;
+          setRoomURL(url);
+          setDemoState('call');
+          setExp(roomExp);
+          setIsError(false);
+        } catch (e) {
+          setDemoState('home');
+          setIsError(true);
+        }
+      }
+    },
+    [roomURL, exp]
+  );
+
+  // Updates state if a room is provided
+  const handleRoomInput = useCallback(
+    (e) => {
+      setRoomURL(e?.target?.value);
+      if (e?.target?.checkValidity()) {
+        setRoomValidity(true);
+        console.log(roomValidity);
```

Comment:
not sure if you left console logs in intentionally but if not, I'd just do a quick cmd-f for them and delete :)

Reply:
All gone, thanks!